Kernel/Firmware/EFI/EFI.h

```diff
@@ -0,0 +1,144 @@
+// e.g. the AllocatePages() "Pages" argument: https://uefi.org/specs/UEFI/2.10/07_Services_Boot_Services.html#efi-boot-services-allocatepages
+static constexpr size_t EFI_PAGE_SIZE = 4 * KiB;
+
+// EFI Data Types: https://uefi.org/specs/UEFI/2.10/02_Overview.html#data-types
+
+// BOOLEAN, Logical Boolean. 1-byte value containing a 0 for FALSE or a 1 for TRUE. Other values are undefined.
+enum class Boolean : u8 {
+    False = 0,
+    True = 1,
+};
+
+// EFI_GUID, 128-bit buffer containing a unique identifier value. Unless otherwise specified, aligned on a 64-bit boundary.
+struct alignas(u64) GUID {
+    u32 part1;
+    u16 part2;
+    u16 part3;
+    u8 part4[8];
+
+    bool operator==(GUID const&) const = default;
+};
+static_assert(AssertSize<GUID, 16>());
+
+// EFI_STATUS, Status code
+// Standard status codes are defined here: https://uefi.org/specs/UEFI/2.10/Apx_D_Status_Codes.html
+static constexpr FlatPtr ERROR_MASK = 1ull << (NumericLimits<FlatPtr>::digits() - 1);
+enum class Status : FlatPtr {
+    Success = 0,
+
+    LoadError = 1 | ERROR_MASK,
+    InvalidParameter = 2 | ERROR_MASK,
+    Unsupported = 3 | ERROR_MASK,
+    BadBufferSize = 4 | ERROR_MASK,
+    BufferTooSmall = 5 | ERROR_MASK,
+    NotReady = 6 | ERROR_MASK,
+    DeviceError = 7 | ERROR_MASK,
+    WriteProtected = 8 | ERROR_MASK,
+    OutOfResources = 9 | ERROR_MASK,
+    VolumeCorrupted = 10 | ERROR_MASK,
+    VolumeFull = 11 | ERROR_MASK,
+    NoMedia = 12 | ERROR_MASK,
+    MediaChanged = 13 | ERROR_MASK,
+    NotFound = 14 | ERROR_MASK,
+    AccessDenied = 15 | ERROR_MASK,
+    NoResponse = 16 | ERROR_MASK,
+    NoMapping = 17 | ERROR_MASK,
+    Timeout = 18 | ERROR_MASK,
+    NotStarted = 19 | ERROR_MASK,
+    AlreadyStarted = 20 | ERROR_MASK,
+    Aborted = 21 | ERROR_MASK,
+    ICMPError = 22 | ERROR_MASK,
+    TFTPError = 23 | ERROR_MASK,
+    ProtocolError = 24 | ERROR_MASK,
+    IncompatibleVersion = 25 | ERROR_MASK,
+    SecurityViolation = 26 | ERROR_MASK,
+    CRCError = 27 | ERROR_MASK,
+    EndOfMedia = 28 | ERROR_MASK,
+
+    EndOfFile = 31 | ERROR_MASK,
+    InvalidLanguage = 32 | ERROR_MASK,
+    CompromisedData = 33 | ERROR_MASK,
+    IPAddressConflict = 34 | ERROR_MASK,
+    HTTPError = 35 | ERROR_MASK,
+};
+
+Optional<StringView> status_description(Status);
+
+// EFI_HANDLE, A collection of related interfaces
+using Handle = FlatPtr;
+
+// EFI_EVENT, Handle to an event structure
+using Event = void*;
+
+// EFI_TPL, Task priority level
+using TPL = FlatPtr;
+
+// EFI_TABLE_HEADER, Data structure that precedes all of the standard EFI table types.
+// https://uefi.org/specs/UEFI/2.10/04_EFI_System_Table.html#id4
+struct TableHeader {
+    u64 signature;
+    u32 revision;
+    u32 header_size;
+    u32 crc32;
+    u32 reserved;
+};
+static_assert(AssertSize<TableHeader, 24>());
+
+// EFI_TIME: https://uefi.org/specs/UEFI/2.10/08_Services_Runtime_Services.html#gettime
+struct Time {
```

Comment:
Does this need `packed` and/or an `AssertSize`?

Reply:
Oh oops, I indeed forgot to add them in multiple of places.

The UEFI spec uses C-like padding rules for their struct definitions, so I don't think we will ever have any problems with things like unwanted padding bytes.
But just to be extra safe I added `AssertSize`s everywhere.